test/SILOptimizer/opt-remark-generator.swift

```diff
@@ -177,66 +177,66 @@ func castAsQuestionDiamondGEP(x: KlassPair) -> SubKlass? {
 
     y.doSomething()
     // We eliminate the rhs retain/release.
-    return y // expected-remark {{retain}}
+    return y // expected-remark {{retain of type 'Klass'}}
              // expected-note @-8:31 {{of 'x.lhs'}}
 }
 
 // We don't handle this test case as well.
 func castAsQuestionDiamondGEP2(x: KlassPair) {
-    switch (x.lhs as? SubKlass, x.rhs as? SubKlass) { // expected-remark @:19 {{retain}}
+    switch (x.lhs as? SubKlass, x.rhs as? SubKlass) { // expected-remark @:19 {{retain of type 'Klass'}}
                                                       // expected-note @-2 {{of 'x.lhs'}}
-                                                      // expected-remark @-2:39 {{retain}}
+                                                      // expected-remark @-2:39 {{retain of type 'Klass'}}
                                                       // expected-note @-4 {{of 'x.rhs'}}
     case let (.some(x1), .some(x2)):
-        print(x1, x2) // expected-remark {{retain}}
-                      // expected-remark @-1 {{retain}}
-                      // expected-remark @-2 {{release}}
-                      // expected-remark @-3 {{release}}
-                      // expected-remark @-4 {{release}}
+        print(x1, x2) // expected-remark {{retain of type 'Optional<SubKlass>'}}
+                      // expected-remark @-1 {{retain of type 'Optional<SubKlass>'}}
+                      // expected-remark @-2 {{release of type}}
+                      // expected-remark @-3 {{release of type 'Optional<SubKlass>'}}
+                      // expected-remark @-4 {{release of type 'Optional<SubKlass>'}}
     case let (.some(x1), nil):
-        print(x1) // expected-remark {{retain}}
-                  // expected-remark @-1 {{release}}
-                  // expected-remark @-2 {{release}}
+        print(x1) // expected-remark {{retain of type 'SubKlass'}}
```

Comment:
Hi @gottesmm,

Since the introduction of these changes the Android ARMv7 CI has been failing. The problem is this message and the similar message below for `x2`. Instead of `retain of type 'SubKlass'` I see `retain of type 'Optional<SubKlass>'` (like the one in line 191 above). This only happens in ARMv7, and not AArch64, so it can be a difference in 32 bits vs. 64 bits. I have also seen cases in which Android ARMv7 seems to not apply some optimizations that AArch64 does.

I know it is a strange question, but do you have any idea of what can be happening here? If there's an easy fix it would be great to apply it. Otherwise I might XFAIL the test.

Thanks!